tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/graal/compiler/TornadoSketchTier.java

```diff
@@ -83,6 +84,7 @@ public TornadoSketchTier(OptionValues options, CanonicalizerPhase.CustomSimplifi
         appendPhase(new TornadoAutoParalleliser());
         appendPhase(new TornadoDataflowAnalysis());
         appendPhase(new TornadoPanamaPrivateMemory());
+        appendPhase(new TornadoBatchFunctionAnalysis());
```

Comment:
This phase will be invoked for all cases (even when batch processing is not enabled), right?

Reply:
It seems there is a check inside the phase to prevent from running it everytime.

Reply:
which check is this?

Reply:
This phase in the sketcher just checks if the loop index is directly written into the output buffer and it does not perform any changes in the graph. Since we do not pass any information related to batches to the sketches phases, this check is always performed. However, the phase that actually alters the graph (`TornadoBatchGlobalIndexOffset`) checks if there is batch processing before applying any changes 

```
if (batchSize == 0) {
            return;
}
```
If the `batchSize` has not been set, there is no batch processing.

Reply:
The check inside the phase is correct. We want to append all phases to the SketchTier, and then select it using runtime information.

Reply:
excellent, thanks @mairooni. I was mislead and looking into the `TornadoBatchFunctionAnalysis` class.